mhkit/acoustics/analysis.py

```diff
@@ -0,0 +1,806 @@
+    water_depth: int, float or array-like
+        Depth of the water column in meters.
+    c: float, optional
+        Speed of sound in the water column in meters per second. Default is 1500 m/s.
+    c_seabed: float, optional
+        Speed of sound in the seabed material in meters per second. Default is 1700 m/s.
+
+    Returns
+    -------
+    f_min: float or numpy.ndarray
+        The minimum cutoff frequency in Hz.
+
+    Reference
+    ---------
+    Jennings 2011 - Computational Ocean Acoustics, 2nd ed.
+    """
+
+    # Convert water_depth to a NumPy array for vectorized operations
+    water_depth = np.asarray(water_depth)
+
+    # Validate water_depth
+    if not np.issubdtype(water_depth.dtype, np.number):
+        raise TypeError("'water_depth' must be a numeric type or array of numerics.")
+
+    if not isinstance(c, (int, float)):
+        raise TypeError("'c' must be a numeric type (int or float).")
+    if not isinstance(c_seabed, (int, float)):
+        raise TypeError("'c_seabed' must be a numeric type (int or float).")
+
+    if np.any(water_depth <= 0):
+        raise ValueError("All elements of 'water_depth' must be positive numbers.")
+    if c <= 0:
+        raise ValueError("'c' must be a positive number.")
+    if c_seabed <= 0:
+        raise ValueError("'c_seabed' must be a positive number.")
+    if c_seabed <= c:
+        raise ValueError("'c_seabed' must be greater than 'c'.")
+
+    fmin = c / (4 * water_depth * np.sqrt(1 - (c / c_seabed) ** 2))
+
+    return fmin
+
+
+def sound_pressure_spectral_density(
+    pressure: xr.DataArray, fs: Union[int, float], bin_length: Union[int, float] = 1
+) -> xr.DataArray:
+    """
+    Calculates the mean square sound pressure spectral density from audio
+    samples split into FFTs with a specified bin length in seconds, using Hanning
+    windowing with 50% overlap. The amplitude of the PSD is adjusted
+    according to Parseval's theorem.
+
+    Parameters
+    ----------
+    pressure: xarray.DataArray (time)
+        Sound pressure in [Pa] or voltage [V]
+    fs: int or float
+        Data collection sampling rate [Hz]
+    bin_length: int or float
+        Length of time in seconds to create FFTs. Default: 1.
+
+    Returns
+    -------
+    spsd: xarray.DataArray (time, freq)
+        Spectral density [Pa^2/Hz] indexed by time and frequency
+    """
+
+    # Type checks
+    if not isinstance(pressure, xr.DataArray):
+        raise TypeError("'pressure' must be an xarray.DataArray.")
+    if not isinstance(fs, (int, float)):
+        raise TypeError("'fs' must be a numeric type (int or float).")
+    if not isinstance(bin_length, (int, float)):
+        raise TypeError("'bin_length' must be a numeric type (int or float).")
+
+    # Ensure that 'pressure' has a 'time' coordinate
+    if "time" not in pressure.dims:
+        raise ValueError("'pressure' must be indexed by 'time' dimension.")
+
+    # window length of each time series
+    nbin = bin_length * fs
```

Comment:
If bin and fs can be floats I think this could be a float. Maybe best practice to cast as int?

Reply:
"nbin" doesn't have to be a float, and it won't throw an error with the binning functions used here.